packages/astro/src/core/app/node.ts

```diff
@@ -85,11 +85,11 @@ export class NodeApp extends App {
 		// @example "443,8080,80" => "443"
 		const forwardedPort = getFirstForwardedValue(req.headers['x-forwarded-port']);
 		const port =
-			forwardedPort ?? req.socket?.remotePort?.toString() ?? (isEncrypted ? '443' : '80');
+			forwardedPort ??  (isEncrypted ? '443' : '80');
```

Comment:
Doesn't this still mean that the port will always have a value if `forwardedPath` is `undefined`? So the part still gets added to the `hostnamePort` and in turn `Astro.url`?

Reply:
I reverted the logic to how it was before https://github.com/withastro/astro/pull/12130. Before, `port` was computed from  the `x-forwarded-port` header. The new change computes the port using the new algorithm, however `port` is still `undefined`, hence we use an inline condition.

This should fix the issue